options/rtld/generic/linker.cpp

```diff
@@ -927,9 +956,22 @@ void doInitialize(SharedObject *object) {
 	__ensure(object->wasLinked);
 	__ensure(!object->wasInitialized);
 
-	// if the object has dependencies we initialize them first
-	for(size_t i = 0; i < object->dependencies.size(); i++)
-		__ensure(object->dependencies[i]->wasInitialized);
+	// If the object has dependencies we initialize them first
```

Comment:
Is this (and down below) really the place we should be checking for circular dependencies? Seems a bit annoying and violates DRY. Isn't it better to consider this an invariant of loaded objects, and validate at object construction time the invariant holds?

Reply:
This is part of #1092

Reply:
Ok, done